ddtrace/contrib/graphql/patch.py

```diff
@@ -0,0 +1,264 @@
+            _set_span_errors(result.errors, span)
+        return result
+
+
+def _init_span(span):
+    # type: (Span) -> None
+    """mark span as measured and set sample rate"""
+    span.set_tag(SPAN_MEASURED_KEY)
+    sample_rate = config.graphql.get_analytics_sample_rate()
+    if sample_rate is not None:
+        span.set_tag(ANALYTICS_SAMPLE_RATE_KEY, sample_rate)
+
+
+def _resolver_middleware(next_middleware, root, info, **args):
+    """
+    trace middleware which wraps the resolvers of graphql fields.
+    Note - graphql middlewares can not be a partial. It must be a class or a function.
+    """
+    pin = Pin.get_from(graphql)
+    if not pin or not pin.enabled():
+        return next_middleware(root, info, **args)
+
+    with pin.tracer.trace(
+        name="graphql.resolve",
+        resource=info.field_name,
+        service=trace_utils.int_service(pin, config.graphql),
+        span_type=SpanTypes.GRAPHQL,
+    ):
+        return next_middleware(root, info, **args)
+
+
+def _inject_trace_middleware_to_args(trace_middleware, args, kwargs):
+    # type: (Callable, Tuple, Dict) -> Tuple[Tuple, Dict]
+    """
+    Adds a trace middleware to graphql.execute(..., middleware, ...)
+    """
+    middlewares_arg = 8
+    if graphql_version >= (3, 2):
+        # middleware is the 10th argument graphql.execute(..) version 3.2+
+        middlewares_arg = 9
+
+    # get middlewares from args or kwargs
+    try:
+        middlewares = get_argument_value(args, kwargs, middlewares_arg, "middleware") or []
+        # if middleware is a MiddlewareManager get iterable from field
+        if isinstance(middlewares, MiddlewareManager):
+            middlewares = middlewares.middlewares  # type: Iterable
+    except ArgumentError:
+        middlewares = []
+
+    # add trace_middleware to the list of execute middlewares
+    middlewares = [trace_middleware] + list(middlewares)
+
+    # update args and kwargs to contain trace_middleware
+    args, kwargs = set_argument_value(args, kwargs, middlewares_arg, "middleware", middlewares)
+    return args, kwargs
+
+
+def _get_source_str(obj):
+    # type: (Union[str, Source, Document]) -> str
+    """
+    Parses graphql Documents and Source objects to retrieve
+    the graphql source input for a request.
+    """
+    if isinstance(obj, str):
+        source_str = obj
+    elif isinstance(obj, Source):
+        source_str = obj.body
+    elif isinstance(obj, Document):
+        source_str = obj.loc.source.body
+    else:
+        source_str = ""
+    # remove new lines, tabs and extra whitespace from source_str
+    return re.sub(r"\s+", " ", source_str).strip()
+
+
+def _set_span_errors(errors, span):
+    # type: (List[GraphQLError], Span) -> None
+    if not errors:
+        # do nothing if the list of graphql errors is empty
+        return
+
+    error_msgs = "\n".join([stringify(error) for error in errors])
+    span.set_exc_fields(GraphQLError, error_msgs, "")
```

Comment:
we can just import the error meta keys and set the tags ourselves here rather than introduce a method on span.

Although this helper does seem useful, we just don't need to introduce it here.